File: packages/python/examples/lifecycle_review/schemas.py
from __future__ import annotations

from pydantic import BaseModel, Field


class CaseInput(BaseModel):
    case_id: str = Field(description="Case identifier.")
    customer_name: str = Field(description="Customer name.")
    request: str = Field(description="Customer request to review.")
    risk_notes: list[str] = Field(default_factory=list, description="Known risk notes.")


class RiskAssessment(BaseModel):
    case_id: str
    risk_level: str
    summary: str
    flags: list[str] = Field(default_factory=list)


class ReviewPacket(BaseModel):
    case_id: str
    recommendation: str
    summary: str
    approval_required: bool
    flags: list[str] = Field(default_factory=list)


class FinalDecision(BaseModel):
    case_id: str
    decision: str
    summary: str
    approved: bool

File: packages/typescript/temporal-yaml/examples/lifecycle-review/schemas.ts
/**
 * Zod ports of the Python example's pydantic models
 * (`packages/python/examples/lifecycle_review/schemas.py`).
 */

import { z } from "zod";

export const CaseInput = z.object({
  case_id: z.string().describe("Case identifier."),
  customer_name: z.string().describe("Customer name."),
  request: z.string().describe("Customer request to review."),
  risk_notes: z.array(z.string()).default([]).describe("Known risk notes."),
});
export type CaseInput = z.infer<typeof CaseInput>;

export const RiskAssessment = z.object({
  case_id: z.string(),
  risk_level: z.string(),
  summary: z.string(),
  flags: z.array(z.string()).default([]),
});
export type RiskAssessment = z.infer<typeof RiskAssessment>;

export const ReviewPacket = z.object({
  case_id: z.string(),
  recommendation: z.string(),
  summary: z.string(),
  approval_required: z.boolean(),
  flags: z.array(z.string()).default([]),
});
export type ReviewPacket = z.infer<typeof ReviewPacket>;

export const FinalDecision = z.object({
  case_id: z.string(),
  decision: z.string(),
  summary: z.string(),
  approved: z.boolean(),
});
export type FinalDecision = z.infer<typeof FinalDecision>;

export const schemas = {
  "schemas:CaseInput": CaseInput,
  "schemas:RiskAssessment": RiskAssessment,
  "schemas:ReviewPacket": ReviewPacket,
  "schemas:FinalDecision": FinalDecision,
};
